src/modules/auth/auth.controller.ts

```diff
@@ -1,27 +1,87 @@
-import { Body, Controller, Post } from '@nestjs/common';
+import {
+  Body,
+  Controller,
+  Get,
+  Post,
+  Req,
+  Res,
+  UnauthorizedException,
+} from '@nestjs/common';
 import { RegisterDto } from './dto/request/register.dto';
 import { AuthService } from './auth.service';
 import { LoginDto } from './dto/request/login.dto';
+import { Request, Response } from 'express';
+import { ConfigService } from '@nestjs/config';
+import { REFRESH_TOKEN_COOKIE_AGE } from 'src/shared/constants';
 
 @Controller('auth')
 export class AuthController {
-  constructor(private readonly authService: AuthService) {}
+  constructor(
+    private readonly authService: AuthService,
+    private readonly configService: ConfigService,
+  ) {}
 
   @Post('register')
-  async register(@Body() registerDto: RegisterDto) {
-    const accessToken = await this.authService.register(registerDto);
+  async register(
+    @Body() registerDto: RegisterDto,
+    @Res({ passthrough: true }) res: Response,
+  ) {
+    const tokens = await this.authService.register(registerDto);
+
+    res.cookie('refreshToken', tokens.refreshToken, {
+      httpOnly: true,
+      secure: this.configService.get('NODE_ENV') === 'production',
+      maxAge: REFRESH_TOKEN_COOKIE_AGE,
```

Comment:
you can set this using a library called "ms" which converts strings like 7d, 15m, 15s to milliseconds. So you can get AUTH_REFRESH_TOKEN_EXPIRES_IN variable and pass it to ms and get milliseconds.

Reply:
fixed